lib/parsers/csv_layout/barcode_creatable_parser.rb

```diff
@@ -8,15 +8,17 @@ def updater
       end
 
       def asset
-        @instance ||= Asset.find_or_import_asset_with_barcode(barcode)
-        unless @instance
-          @instance = Asset.new(barcode: barcode)
-          @instance.generate_uuid!
-          updater.create_assets([@instance])
-          updater.add(@instance, 'barcode', barcode)
-          updater.add(@instance , 'a', 'Tube')
+        @asset ||= @parser.find_or_import_asset_with_barcode(barcode) || generate_asset
```

Comment:
By delegating to the parser we can build a cache of all barcodes in the file, reducing the number of database queries we need to make considerably.